jni/lib/src/jni_array.dart

```diff
@@ -0,0 +1,292 @@
+    final it = iterable.skip(skipCount).take(size);
+    _allocate(sizeOf<JInt>() * size, (ptr) {
+      it.forEachIndexed((index, element) {
+        ptr[index] = element;
+      });
+      _env.SetIntArrayRegion(reference, start, size, ptr);
+    });
+  }
+}
+
+extension LongJniArray on JniArray<JLong> {
+  int operator [](int index) {
+    return elementAt(index, JniType.longType).long;
+  }
+
+  void operator []=(int index, int value) {
+    RangeError.checkValidIndex(index, this);
+    _allocate(sizeOf<JLong>(), (ptr) {
+      ptr.value = value;
+      _env.SetLongArrayRegion(reference, index, 1, ptr);
+    });
+  }
+
+  void setRange(int start, int end, Iterable<int> iterable,
+      [int skipCount = 0]) {
+    RangeError.checkValidRange(start, end, length);
+    final size = end - start;
+    final it = iterable.skip(skipCount).take(size);
+    _allocate(sizeOf<JLong>() * size, (ptr) {
+      it.forEachIndexed((index, element) {
+        ptr[index] = element;
+      });
+      _env.SetLongArrayRegion(reference, start, size, ptr);
+    });
+  }
+}
+
+extension FloatJniArray on JniArray<JFloat> {
+  double operator [](int index) {
+    return elementAt(index, JniType.floatType).float;
+  }
+
+  void operator []=(int index, double value) {
+    RangeError.checkValidIndex(index, this);
+    _allocate(sizeOf<JFloat>(), (ptr) {
+      ptr.value = value;
+      _env.SetFloatArrayRegion(reference, index, 1, ptr);
+    });
+  }
+
+  void setRange(int start, int end, Iterable<double> iterable,
+      [int skipCount = 0]) {
+    RangeError.checkValidRange(start, end, length);
+    final size = end - start;
+    final it = iterable.skip(skipCount).take(size);
+    _allocate(sizeOf<JFloat>() * size, (ptr) {
+      it.forEachIndexed((index, element) {
+        ptr[index] = element;
+      });
+      _env.SetFloatArrayRegion(reference, start, size, ptr);
+    });
+  }
+}
+
+extension DoubleJniArray on JniArray<JDouble> {
+  double operator [](int index) {
+    return elementAt(index, JniType.doubleType).doubleFloat;
+  }
+
+  void operator []=(int index, double value) {
+    RangeError.checkValidIndex(index, this);
+    _allocate(sizeOf<JDouble>(), (ptr) {
+      ptr.value = value;
+      _env.SetDoubleArrayRegion(reference, index, 1, ptr);
+    });
+  }
+
+  void setRange(int start, int end, Iterable<double> iterable,
+      [int skipCount = 0]) {
+    RangeError.checkValidRange(start, end, length);
+    final size = end - start;
+    final it = iterable.skip(skipCount).take(size);
+    _allocate(sizeOf<JDouble>() * size, (ptr) {
+      it.forEachIndexed((index, element) {
+        ptr[index] = element;
+      });
+      _env.SetDoubleArrayRegion(reference, start, size, ptr);
+    });
+  }
+}
+
+extension ObjectJniArray<T extends JniObject> on JniArray<T> {
+  JniObject operator [](int index) {
+    return JniObject.fromRef(elementAt(index, JniType.objectType).object);
+  }
+
+  void operator []=(int index, JniObject value) {
```

Comment:
Unlike `elementAt` the logic of setting is not the same between primitives and objects yet (since I did not create an accessor for it).

We could even write`setRange(index, index + 1, [value])` without the need of any other function if you prefer.

Reply:
But that's going to introduce a tiny bit of overhead compared to what we currently have.

Reply:
More than tiny bit of overhead, there's allocation :(.

An idea might be have one accessor for objects, one for all integer types taking long, and use downcasting in C. That will reduce allocation burden in single set.

Reply:
Here's the `setRange` for Objects:

```dart
  void setRange(int start, int end, Iterable<JniObject> iterable,
      [int skipCount = 0]) {
    RangeError.checkValidRange(start, end, length);
    final size = end - start;
    final it = iterable.skip(skipCount).take(size);
    it.forEachIndexed((index, element) {
      this[index] = element;
    });
  }
  ```

It is simply calling `operator []=` for each element. Creating the extra list is the overhead, but that's it.